Client/Frontend/NTP/NTPLearnMoreViewController.swift

```diff
@@ -0,0 +1,150 @@
+            
+            let tos = Strings.termsOfServiceURL
+            let tosPart = String(format: Strings.NTP.turnRewardsTos, tos)
+            
+            let hideImages = Strings.NTP.hideSponsoredImages
+            let hideImagesPart = String(format: Strings.NTP.chooseToHideSponsoredImages, hideImages)
+            
+            let fullBodyText = "\(tosPart) \(hideImagesPart)"
+            
+            config.bodyText =
+                (text: fullBodyText,
+                 urlInfo: [tos: "tos", hideImages: "hide-sponsored-images"],
+                 action: { [weak self] action in
+                    if action.absoluteString == "tos" {
+                        let urlString = "https://www.brave.com/terms_of_use"
+                        guard let url = URL(string: urlString) else { return }
+                        self?.showSFSafariViewController(url: url)
+                    } else if action.absoluteString == "hide-sponsored-images" {
+                        Preferences.NewTabPage.backgroundSponsoredImages.value = false
+                        self?.close()
+                    }
+                })
+            
+            config.primaryButtonConfig =
+                (text: Strings.NTP.turnOnBraveRewards,
+                 showCoinIcon: false,
+                 action: { [weak self] in
+                    guard let rewards = (UIApplication.shared.delegate as? AppDelegate)?
+                        .browserViewController.rewards else { return }
+                    
+                    rewards.ledger.createWalletAndFetchDetails { _ in }
+                    self?.close()
+                })
+            
+            config.secondaryButtonConfig =
+                (text: Strings.disclaimerLearnMore,
+                 action: { [weak self] in
+                    guard let url = URL(string: "https://brave.com/brave-rewards/") else { return }
+                    self?.showSFSafariViewController(url: url)
+                })
+        case .getPaidTurnAdsOn:
+            config.headerText = Strings.NTP.getPaidForThisImageTurnAds
+            
+            let hideImages = Strings.NTP.hideSponsoredImages
+            let hideImagesPart = String(format: Strings.NTP.chooseToHideSponsoredImages, hideImages)
+            
+            config.bodyText =
+                (text: hideImagesPart,
+                 urlInfo: [hideImages: "hide-sponsored-images"],
+                 action: { [weak self] action in
+                    if action.absoluteString == "hide-sponsored-images" {
+                        Preferences.NewTabPage.backgroundSponsoredImages.value = false
+                        self?.close()
+                    }
+                })
+            
+            config.primaryButtonConfig =
+                (text: Strings.NTP.turnOnBraveAds,
+                 showCoinIcon: true,
+                 action: { [weak self] in
+                    guard let rewards = (UIApplication.shared.delegate as? AppDelegate)?
```

Comment:
Same as above. Perhaps we should consider making a rewards _singleton_ 😱

Reply:
NO :D
I just didn't want to pass it to each view so many times, but after refactoring all these notification types are now abstracted in just two classes we should be good

Reply:
I'm only thinking of decoupling rewards from BVC, move it to `Profile` or other object that can be injected